src/lib/util.js

```diff
@@ -2,6 +2,32 @@ import config from 'config';
 import crypto from 'crypto';
 const algorithm = 'aes-256-ctr';
 
+/**
+ * Get current store code from parameter passed from the vue storefront frotnend app
+ * @param {Express.Request} req 
+ */
+export function getCurrentStoreCode (req) {
+  if (req.headers['x-vs-store-code']) {
+    return req.headers['x-vs-store']
+  }
+  if (req.query.storeCode) {
+    return req.query.storeCode
+  }
+  return null
+}
+
+/**
+ * Get the config.storeViews[storeCode]
+ * @param {string} storeCode 
+ */
+export function getCurrentStoreView (storeCode = null) {
+  if (storeCode && config.storeViews[storeCode]) {
+    storeView = config.storeViews[storeCode]
```

Comment:
Why aren't we returning the storeView here? 

For me it looks like a bug here and the default that we return looks bad to in my eyes. Here we should do it like we do in prepareStoreView in the frontend.

```javascript
  let storeView = { // current, default store
    tax: config.tax,
    i18n: config.i18n,
    elasticsearch: config.elasticsearch,
    storeCode: null,
    storeId: config.defaultStoreCode && config.defaultStoreCode !== '' ? config.storeViews[config.defaultStoreCode].storeId : 1
  }
```

Reply:
Thanks fixed